grpcutil/dns_resolver.go

```diff
@@ -251,7 +261,7 @@ func (w *dnsWatcher) lookupHost() map[string]*Update {
 func (w *dnsWatcher) lookup() []*Update {
 	newAddrs := w.lookupSRV()
 	if newAddrs == nil {
-		// If failed to get any balancer address (either no corresponding SRV for the
+		// If we failed to get any balancer address (either no corresponding SRV for the
```

Comment:
If we're changing this line, we should remove mention of balancer (this code was doing loadbalancer lookup originally).

Reply:
Note that there are other refences to GRPCLB in this code, that should be removed too.

Reply:
@pstibrany Isn't looking up the SRV record a load balancing strategy? I.e. load balancing by picking one of the SRV record's targets?

Reply:
@pstibrany If we want to remove other GRPCLB references, can we do that in another PR? Otherwise I feel as if we're mixing separate concerns.

Reply:
It's a matter of usage... In Cortex we don't use it for load-balancing, but to discover *all* targets to connect to.

Reply:
> @pstibrany If we want to remove other GRPCLB references, can we do that in another PR? Otherwise I feel as if we're mixing separate concerns.

This PR started by removing "grpclb" string, so we may as well continue in that direction.

Reply:
> It's a matter of usage... In Cortex we don't use it for load-balancing, but to discover all targets to connect to.

@pstibrany thanks for clarifying that.

I don't agree the PR is about removing the grpclb string though. It's about dropping the assumption that the service part of the SRV record is "grplb". If we remove GRPCLB references we'll reduce the public interface (of `Update`, which gets returned by `Watcher.Next`).

Reply:
I don't know if the `AddrMetadataGRPCLB` type is useful in any way, but I'd rather deal with that separately.

Reply:
My take is that we started with removal of hardcoded assumption that we *always* want to do "grpclb" service resolution, and instead may want to do other services too. However there is still lot of grpc-related naming in this package. I think we should drop the following:
- `AddressType`, `Default` and `GRPCLB` constants
- `AddrMetadataGRPCLB` (if we remove `AddressType`, there is only `ServerName` left, with comment "ServerName is the name of the grpc load balancer. Used for authentication." which makes no sense here.
- `Metadata` field from `Update`

I would stop there for this PR. WDYT?

Reply:
@pstibrany I guess that makes sense. What do youj mean by the `Default` constant though? I dont' see any.

Reply:
> @pstibrany I guess that makes sense. What do youj mean by the `Default` constant though? I don't see any.

Sorry, that's a typo. I meant `Backend`.

Reply:
@pstibrany I removed the gRPC LB related code. Look good?